pkg/smartcontract/manifest/parameter.go

```diff
@@ -79,42 +80,28 @@ func (p Parameters) AreValid() error {
 			return fmt.Errorf("parameter #%d/%q: %w", i, p[i].Name, err)
 		}
 	}
-	if len(p) < 2 {
-		return nil
-	}
-	names := make([]string, len(p))
-	for i := range p {
-		names[i] = p[i].Name
-	}
-	if stringsHaveDups(names) {
+	if sliceHasDups(p, func(a, b Parameter) int {
+		return cmp.Compare(a.Name, b.Name)
+	}) {
 		return errors.New("duplicate parameter name")
 	}
 	return nil
 }
 
-// stringsHaveDups checks the given set of strings for duplicates. It modifies the slice given!
-func stringsHaveDups(strings []string) bool {
-	slices.Sort(strings)
-	for i := range strings {
-		if i == 0 {
-			continue
-		}
-		if strings[i] == strings[i-1] {
-			return true
-		}
+// sliceHasDups checks the slice for duplicate elements.
+func sliceHasDups[S ~[]E, E any](x S, cmp func(a, b E) int) bool {
+	if len(x) < 2 {
+		return false
 	}
-	return false
-}
-
-// permissionDescsHaveDups checks the given set of strings for duplicates. It modifies the slice given!
-func permissionDescsHaveDups(descs []PermissionDesc) bool {
-	slices.SortFunc(descs, PermissionDesc.Compare)
-	for i := range descs {
+	if len(x) > 2 {
+		x = slices.Clone(x)
+		slices.SortFunc(x, cmp)
+	}
+	for i := range x {
 		if i == 0 {
 			continue
 		}
-		j := i - 1
-		if descs[i].Compare(descs[j]) == 0 {
+		if cmp(x[i-1], x[i]) == 0 {
 			return true
 		}
 	}
```

Comment:
This cycle is not needed, there are only two elements. Just use `return cmp(x[0], x[1])`.

Reply:
It can have more of them.

Reply:
OK, I missed there's no `return` after sorting code above.